test/dynamo/test_modules.py

```diff
@@ -2676,6 +2676,8 @@ def debug_compiler(gm, _):
         class TestModule(torch.nn.Module):
             def __init__(self, buf) -> None:
                 super().__init__()
+                # Changing this one to nn.Buffer fails because `nn.Buffer` does a .detach()
+                # so the value in self.tx.output.side_effects will no longer evaluate to True
```

Comment:
Since `nn.Buffer` calls detach, this line will no longer evaluate to True, I am not sure whether this is the intended behavior of this `mark_static` logic, or whether we should make `nn.Buffer` work with this if the tensor passed to the `nn.Buffer` constructor had previously been seen 

https://github.com/pytorch/pytorch/blob/8fe5b93667b60e37c12d288659a25cbd5ae53c79/torch/_dynamo/variables/builder.py#L1197